crates/oxc_linter/src/disable_directives.rs

```diff
@@ -604,3 +693,169 @@ semi*/
             .test();
     }
 }
+
+#[cfg(test)]
+mod test_unused_directives {
+    use oxc_allocator::Allocator;
+    use oxc_ast::Comment;
+    use oxc_parser::Parser;
+    use oxc_semantic::{Semantic, SemanticBuilder};
+    use oxc_span::SourceType;
+
+    use super::{DisableDirectivesBuilder, DisabledRule};
+
+    fn process_source<'a>(allocator: &'a Allocator, source_text: &'a str) -> Semantic<'a> {
+        let source_type = SourceType::default();
+        let parser_ret = Parser::new(allocator, source_text, source_type).parse();
+        let semantic_ret = SemanticBuilder::new().build(allocator.alloc(parser_ret.program));
+        semantic_ret.semantic
+    }
+
+    #[test]
+    fn unused_enable_all() {
+        let allocator = Allocator::default();
+        let semantic = process_source(
+            &allocator,
+            r"
+            console.log();
+            /* eslint-enable */
+            console.log();
+            ",
+        );
+        let comments = semantic.comments();
+        let directives = DisableDirectivesBuilder::new().build(semantic.source_text(), comments);
+
+        let unused = directives.unused_enable_comments();
+
+        assert_eq!(
+            unused.first().map(|(_, span)| *span),
+            comments.first().map(Comment::content_span)
+        );
+    }
+
+    #[test]
+    fn unused_enable_rules() {
+        let allocator = Allocator::default();
+        let semantic = process_source(
+            &allocator,
+            r"
+            console.log();
+            /* eslint-enable no-debugger, no-console */
+            console.log();
+            ",
+        );
+        let comments = semantic.comments();
+        let directives = DisableDirectivesBuilder::new().build(semantic.source_text(), comments);
+
+        let unused = directives.unused_enable_comments();
+
+        assert_eq!(unused.len(), 2);
+        assert_eq!(
+            unused.first().map(|(_, span)| *span),
+            comments.first().map(Comment::content_span)
+        );
+        assert_eq!(
+            unused.last().map(|(_, span)| *span),
+            comments.first().map(Comment::content_span)
+        );
```

Comment:
this goes for all of the other tests in this file where there is a rule name, we should check the rule name too, not just the comment span:
```suggestion
        assert_eq!(unused.len(), 2);
        let comment_span = comments.first().unwrap().content_span();
        let disable_debugger = unused.first().unwrap();
        let disable_console = unused.last().unwrap();
        assert_eq!(disable_debugger.span, comment_span);
        assert_eq!(disable_debugger.rule_name, "no-debugger");
        assert_eq!(disable_console.span, comment_span);
        assert_eq!(disable_console.rule_name, "no-console"
```